### DISCUSSION TITLE
Can major upgrades be to the lowest (next) version rather than highest available?

### DISCUSSION THREAD

**Discussion by georgekaz** [renovatebot/renovate#32199] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Github, 8.4.0 CE Docker

### Please tell us more about your question or problem

Is there a setting which would allow for creation of a single PR for major version upgrades to the lowest next version, similar to how https://docs.renovatebot.com/configuration-options/#vulnerabilityfixstrategy allows for "lowest" as an option?

At the moment, you can achieve getting a PR for the next major version if you set https://docs.renovatebot.com/configuration-options/#separatemultiplemajor to "true", but you'll simultaneously get PRs for all major version upgrades. If you set this option to "false" you only get a single PR but it's always for the highest major version available.

If would be useful if the equivalent to vulnerabilityfixstrategy was available for major upgrades too. Someone else requested similar for minor upgrades [here](https://github.com/renovatebot/renovate/discussions/28392), so maybe a more generic upgradeStrategy setting for packageRules is what I'm asking about. Does this exist and I've missed it?

Thanks

**Comment by RahulGautamSingh** [comment, marked as answer]
No, we don't have configuration option for this.

**Reply by RahulGautamSingh** [reply]
If we try to implement this, then it would go in the lookup logic. We filter versions there and select a list of satisfying verisons from within which we select the latest ... maybe we can pass a "lowest" option there and since it will be in an array, after sorting we can pick the one next to the current version. (there could be more details I might be missing but this is the overview)

It's not an priority right now, so we cannot promise a timeline or an issue. But, if you'd like to contribute an implementation, you're welcome.

**Reply by georgekaz** [reply]
Thanks for the answer Rahul. Do you mean the logic needs to go [here](https://github.com/renovatebot/renovate/blob/main/lib/workers/repository/process/lookup/index.ts)? I'll see what I can do about contributing.

**Reply by RahulGautamSingh** [reply]
Yes
